### PR TITLE
Adding dark mode with button that toggles additional css

### PR DESCRIPTION
**Pull request by jannikbuhr**
You knew this was coming... Join the dark side, we have cookies! (No really, the darkmode uses `localStorage` to remember a users preferences)

### CODE REVIEW

File: ui/public/darkmode.css [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+body, nav {
+  color: rgb(255, 253, 248);
+  background: rgb(19, 19, 20);
+}
+
+a, .collapsible label, .svelte-gwqplq {
```
Comment (liwde) [inline ui/public/darkmode.css:7]: We shouldn't rely on css classes generated by the framework

File: ui/src/controls/ToggleDarkModeButton.svelte [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{/if}
+</svelte:head>
+
+<script>
+import { createEventDispatcher } from 'svelte';
+const dispatch = createEventDispatcher();
+var stored = localStorage.getItem("dark-mode-storage");
+
+if (!stored) {
+  if (window.matchMedia("(prefers-color-scheme: dark)").matches ) {
```
Comment (liwde) [inline ui/src/controls/ToggleDarkModeButton.svelte:13]: Would we want to listen to the changes? I'm not sure if it's a valid use case, but could imagine iOS to automatically switch themes based on time or sensor input, and I guess a user could expect the site to change without reloading: https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener

You'd have to make sure to de-register the listener, though, once the theme is switched manually. And there is no way back to "system theme" once we switched once... What do you think?

File: ui/src/controls/ToggleDarkModeButton.svelte [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    stored = "dark";
+  } else {
+    stored = "light";
+  }
+}
+
+function switchTheme() {
+    if (stored == "light") {
```
Comment (liwde) [inline ui/src/controls/ToggleDarkModeButton.svelte:21]:
```suggestion
    if (stored === "light") {
```

File: ui/src/controls/ToggleDarkModeButton.svelte [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        stored = "dark";
+    } else {
+        stored = "light";
+      }
```
Comment (liwde) [inline ui/src/controls/ToggleDarkModeButton.svelte:25]: Please use consistent indentation as specified in the editorconfig (two spaces).

Comment (liwde) [reply ui/src/controls/ToggleDarkModeButton.svelte:25]: Also, somewhere here we should set the localStorage

File: ui/src/controls/ToggleDarkModeButton.svelte [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+</script>
+
+<button on:click="{switchTheme}">
+<img src="img/icon_darkmode.svg" alt="dark mode toggle" class="no-responsive no-border"/>
```
Comment (liwde) [inline ui/src/controls/ToggleDarkModeButton.svelte:30]: I'm not completely sure if we should wrap the icon in a button, or if we should let it stand on its own. I'll experiment a bit with that myself and see how it looks in the navbar :)

File: ui/public/darkmode.css (hunk 1 above)
Comment (liwde) [reply ui/public/darkmode.css:7]: I don't know, where this particular one is coming from, but if necessary, we could move it from its respective component to the global css.

Comment (jannikbuhr) [reply ui/public/darkmode.css:7]: I am removing this in the next commit. This was an artifact from a different button implementation that would stay white without this when in darkmode but is no longer necessary.

File: ui/src/controls/ToggleDarkModeButton.svelte [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<svelte:head>
+{#if stored == "dark"}
+<link rel='stylesheet' href='darkmode.css' id="dark-mode-css">
+{/if}
+</svelte:head>
+
+<script>
+import { createEventDispatcher } from 'svelte';
+const dispatch = createEventDispatcher();
```
[the remark this answers is not in the document]
Comment (jannikbuhr) [reply ui/src/controls/ToggleDarkModeButton.svelte]: Done :)

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 5 above)
Comment (jannikbuhr) [reply ui/src/controls/ToggleDarkModeButton.svelte:28]: Maybe with a span tag as in https://github.com/liwde/blinddichten/pull/1/commits/4a501f63306436014bb80975c50d0590f93fcde8?

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 4 above)
Comment (jannikbuhr) [reply ui/src/controls/ToggleDarkModeButton.svelte]: Indentation fixed in https://github.com/liwde/blinddichten/pull/1/commits/4a501f63306436014bb80975c50d0590f93fcde8, isnt' localStorage set in lines 8 to 16?

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 5 above)
Comment (liwde) [reply ui/src/controls/ToggleDarkModeButton.svelte:28]: I've worked out something

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 4 above)
Comment (liwde) [reply ui/src/controls/ToggleDarkModeButton.svelte]: You're just reading from the localStorage, never writing. The variable would better be named "theme"...

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 5 above)
Comment (liwde) [reply ui/src/controls/ToggleDarkModeButton.svelte]: I've committed it, let me know what you think :)

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 2 above)
Comment (jannikbuhr) [reply ui/src/controls/ToggleDarkModeButton.svelte]: The only way to have both the option to switch automatically and remember a manual selection would be with an additional button that allows going back. I think an additional button would counteract the minimalist design principles of the app.

File: ui/src/controls/ToggleDarkModeButton.svelte (hunk 5 above)
Comment (jannikbuhr) [reply ui/src/controls/ToggleDarkModeButton.svelte]: Looks like you already fixed it in your commit https://github.com/liwde/blinddichten/pull/1/commits/f2eaaebf41574044c1e57a801a7777f0b4703740. I like the way it looks.

**Comment by liwde** [comment]
I think we should do a few additional things:

- The white font seems really bright to me. Maybe something a bit darker. I once read a good blog post on designing a dark mode, and I have to agree: "Pure white text on full black backgrounds makes reading long passages of text tiring for the eye." I'm in favor of a relatively black background, but the font must be a bit darkened.
- A few elements are missing: Input boxes (and their placeholder values) are still black, the "random title" button is as well (css filters might be able to do the trick).
- The progress bars are also really bright in comparison.

**Comment by jannikbuhr** [comment]
Sometimes clicking the button doesn't work and throws:
```
index.mjs:175 Uncaught (in promise) TypeError: Cannot read property 'removeChild' of null
    at detach (index.mjs:175)
    at detach_dev (index.mjs:1487)
    at Object.destroy [as d] (errorList.js:28)
    at Object.update [as p] (ToggleDarkModeButton.svelte:23)
    at update (index.mjs:687)
    at flush (index.mjs:656)
```

from the `switchTheme` function but I have not been able to pinpoint exactly when this happens.

**Comment by jannikbuhr** [comment]
All elements should be covered by the dark mode now. Because shadows don't quite work on a dark background, I experimented with a glowing effect instead in the latest commit.

**Pull request merged**